### PR TITLE
Project Labyrinth - Max & Ingela

### PR DESCRIPTION
**Pull request by IngelaL**
Link to my deployed project: https://labyrinth-game.netlify.app/

My reflections on how this week's project turned out: It was a fun but tough project and we had a great team work.

Things I'd like to have clarified or explained in more detail: Would have been great with some more material this week.

### CODE REVIEW

File: code/src/components/GameDescription.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+import styled from 'styled-components'
+
+import { continueGame } from '../reducers/game'
+
+const StartDescription = styled.div`
+  text-align: center;
+  margin-bottom: 30px; 
+  padding: 0 30px 0 30px;
+
+  @media (min-width: 667px) {
```
Comment (karanmann) [inline code/src/components/GameDescription.js:12]: Cool that you added media queries to make the page change as per the resolution

File: code/src/components/GameDescription.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+`
+
+export const GameDescription = ({inputValue}) => {
+  const start = useSelector((store) => store.game.game)
+  const dispatch = useDispatch()
+
+  return (
+    <>
+     <StartDescription>{start.description}</StartDescription>
+        {start.actions?.map((direction, index) => {
```
Comment (karanmann) [inline code/src/components/GameDescription.js:37]: Cool to add '?' to as a conditional so the mapping only takes place when it has an action :)

File: code/src/components/StartGame.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import { useDispatch, useSelector, batch } from 'react-redux'
+import styled from 'styled-components'
+
+import { createPlayer } from '../reducers/game'
+import { game } from 'reducers/game'
+import { GameDescription } from './GameDescription'
+import { WelcomePlayer } from './WelcomePlayer'
+import maze from '../assets/maze.svg'
+ 
+const GameContainer = styled.div`
```
Comment (karanmann) [inline code/src/components/StartGame.js:11]: You could also make another styled component and import the styles here. Just a recommendation.

File: code/src/reducers/game.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
+import { createSlice } from '@reduxjs/toolkit'
+
+
+const initialState = {
+    game: {}, 
+    history: [],
+    username: "",
+
```
Comment (karanmann) [inline code/src/reducers/game.js:8]: This line spacing can be skipped :)

File: code/src/reducers/game.js [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+export const game = createSlice ({
+    name: 'game', initialState,
+    
+    reducers: {
+    saveGame: (state, action) => {
```
Comment (karanmann) [inline code/src/reducers/game.js:15]: We could add tab to indent the reducers

File: code/src/reducers/game.js [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        .then((json) => {
+            dispatch(game.actions.saveGame(json))
+        })
+    }
+}
+
+export const continueGame = (step, username) => {
+    return (dispatch, getState) => {
+        const state = getState()
+        console.log(state)
```
Comment (karanmann) [inline code/src/reducers/game.js:49]: You could remove the console log on line 49 and 50.

File: code/package.json [hunk 7]
```diff
@@ -12,7 +13,10 @@
     "eslint-plugin-react-hooks": "^2.3.0",
     "react": "^16.12.0",
     "react-dom": "^16.12.0",
-    "react-scripts": "^3.3.0"
+    "react-redux": "^7.2.2",
+    "react-scripts": "^3.3.0",
+    "save": "^2.4.0",
```
Comment (karanmann) [inline code/package.json:18]: We were  wondering about this dependency. Were you trying to save the data locally?

**Review by karanmann — APPROVED** [review]
Hej Ingela & Max,

Very clean and readable code. Its very easy to follow. We loved you styling and would encourage you to try material UI on it. Also from looking at the dependencies we can see that you probably tried to save the state locally.  

/ Sandra and Karan

File: code/package.json (hunk 7 above)
Comment (IngelaL) [reply code/package.json:18]: Actually, we don't remember installing this or where it comes from :)

**Comment by IngelaL** [comment]
Thanks for great feedback. Will check out the material UI 🤩

File: code/src/components/StartGame.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+            dispatch(game.actions.addUserName(inputValue))
+            dispatch(createPlayer(inputValue))
+          })      
+    }
+
+return (
+    <>
+    <GameContainer>
+        <HeaderContainer>
+            <Header>La<span style={{color: '#314834'}}>by</span>rinth game</Header>
```
Comment (puj) [inline code/src/components/StartGame.js:94]: Hmmm, I'm looking for a secret message here.  😄 "by"!

**Review by puj — CHANGES_REQUESTED** [review]
Really clean project Max & Ingela!  Very understandable code and UI.  Nicely done!  Small tweaks before approval:
- Indentation
- Remove `console.log`s

**Comment by IngelaL** [comment]
@puj We have now made the changes requested!

File: code/src/components/StartGame.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+            <Form onSubmit={onSubmit}>
+                <Input 
+                    aria-label='text area'
+                    type= 'text' 
+                    onChange={e => setInputValue(e.target.value)}
+                    value={inputValue}
+                    placeholder='Type your username'>
+                </Input>
+                <Button>Start game</Button>
+            </Form>
```
Comment (HIPPIEKICK) [inline code/src/components/StartGame.js:106]: This shouldn't be indented as much

File: code/src/components/StartGame.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+        e.preventDefault()
+        batch(() => {
+            dispatch(game.actions.addUserName(inputValue))
+            dispatch(createPlayer(inputValue))
+          })      
+    }
+
+return (
+    <>
+    <GameContainer>
```
Comment (HIPPIEKICK) [inline code/src/components/StartGame.js:92]:
```
<>
  <GameContainer>
```

File: code/src/reducers/game.js [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        .then((res) => res.json())
+        .then((json) => {
+            dispatch(game.actions.saveGame(json))
+        })
+    }
+}
+
+export const continueGame = (step, username) => {
+    return (dispatch) => {
+                fetch('https://wk16-backend.herokuapp.com/action', {
```
Comment (HIPPIEKICK) [inline code/src/reducers/game.js:48]: This shouldn't be as much indented

Comment (IngelaL) [reply code/src/reducers/game.js]: @HIPPIEKICK Have now changed the indentation.

**Pull request closed without merging**